### PR TITLE
Transfer psylink and abilities

### PR DESCRIPTION
**Pull request by Diakon5**
Draft. Psylink Hediff is currently transfered successfully, adjusted hediff transfer to copy over Hediff's severity(or level if it's a leveled hediff). ~~Ability copying doesn't work yet~~ Ability copying now works. Checking compatibility with Vanilla Psycasts Expanded first.

### CODE REVIEW

**Comment by Diakon5** [comment]
Vanilla Psycasts Expanded breaks the Psylink transfer. Because of that, any pawn with a psylink can never fully transform into an animal.

**Review by Zeracronius — APPROVED** [review]

**Comment by Diakon5** [comment]
Closes #430.
Incompatibilty with Vanilla Psycasts Expanded reduced to a minor state, Now any Psycaster will retain their psylink but loses their psypowers. Reload may solve it, more explicit compatibility on the way.

**Comment by Diakon5** [comment]
Final commit. Now transformation only causes a single letter confirming the pawn has gained their first psylink. No messages for level-ups. Probably the best I can do without Harmony.

File: Source/Pawnmorphs/Esoteria/PawnTransferUtilities.cs [hunk 1]
```diff
@@ -308,7 +308,15 @@ public static void TransferFavor([NotNull] Pawn pawn1, [NotNull] Pawn pawn2)
                 health2.AddHediff(newHediff);
```
Comment (Zeracronius) [inline Source/Pawnmorphs/Esoteria/PawnTransferUtilities.cs:308]: Hediff_Psylink has a public field called "suppressPostAddLetter"
though a bit fiddly, you could try set that to false between making and adding the hediff if psylink
that might well get rid of the last letter

File: Source/Pawnmorphs/Esoteria/PawnTransferUtilities.cs [hunk 2]
```diff
@@ -269,11 +301,69 @@ public static void TransferFavor([NotNull] Pawn pawn1, [NotNull] Pawn pawn2)
...
+                        else if (otherHediff is Hediff_Level newLevelExitsting)
+                        {
+                            newLevelExitsting.SetLevelTo((int)hediff.Severity);
+                        }
+                        else
+                        {
+                            otherHediff.Severity = hediff.Severity;
+                        }
+                    }
+                }
```
Comment (Zeracronius) [inline Source/Pawnmorphs/Esoteria/PawnTransferUtilities.cs:326]: All of this code is basically identical to what is further below.
Could merge this if you handle "Find or Make + Suppress & add" and then handle severity adjustments

**Review by Zeracronius — CHANGES_REQUESTED** [review]

File: Source/Pawnmorphs/Esoteria/PawnTransferUtilities.cs [hunk 3]
```diff
@@ -269,11 +301,69 @@ public static void TransferFavor([NotNull] Pawn pawn1, [NotNull] Pawn pawn2)
...
+                {
+                    if (newLevel is Hediff_Psylink level)
+                    {
+                        level.ChangeLevel((int)(hediff.Severity - level.Severity), false);
+                    }
+                    else
+                    {
+                        newLevel.SetLevelTo((int)hediff.Severity);
+                    }
+>>>>>>> Stashed changes
```
[the remark this answers is not in the document]
Comment (Diakon5) [reply Source/Pawnmorphs/Esoteria/PawnTransferUtilities.cs:357]: I don't know what caused it

**Pull request closed without merging**